server/services/team_service.py

```diff
@@ -176,10 +176,47 @@ def delete_team_project(team_id, project_id):
         project.delete()
 
     @staticmethod
-    def get_all_teams() -> TeamsListDTO:
-        query = Team.query.distinct()
+    def get_all_teams(
+        team_name_filter: str = None,
+        team_role_filter: str = None,
+        member_filter: int = None,
+        member_request_filter: int = None,
+        manager_filter: int = None,
+    ) -> TeamsListDTO:
+
+        query = db.session.query(Team).outerjoin(TeamMembers).outerjoin(ProjectTeams)
+
+        if team_name_filter:
+            query = query.filter(Team.name.contains(team_name_filter))
```

Comment:
@fitoria Do you think it'd be safe to use it to make the name query case insensitive?
`query = query.filter(Team.name.ilike("%{}%".format(team_name_filter)))`